### PR TITLE
LoadDictionaryCommandFix

### PR DESCRIPTION
**Pull request by openbots-ff**
Modified the load dictionary command so that we no longer have to install that dependency. The code is a bit longer now but it actually seems to run much faster. That query was always really slow on the first execution.

### CODE REVIEW

File: taskt/Core/Automation/Commands/LoadDictionaryCommand.cs [hunk 1]
```diff
@@ -83,15 +84,46 @@ public override void RunCommand(object sender)
...
-            //Query required from workbook using OLEDB
-            DatasetCommands dataSetCommand = new DatasetCommands();
-            DataTable requiredData = dataSetCommand.CreateDataTable(@"Provider=Microsoft.ACE.OLEDB.12.0;Data Source=" + vFilePath + @";Extended Properties=""Excel 12.0;HDR=YES;IMEX=1""", "Select " + v_KeyColumn + "," + v_ValueColumn + " From [" + v_SheetName + "$]");
+            var excelWorkbook = newExcelSession.Workbooks.Open(vFilePath);
+            var excelSheet = excelWorkbook.Sheets[vSheetName];
+ 
+            Microsoft.Office.Interop.Excel.Range last = excelSheet.Cells.SpecialCells(Microsoft.Office.Interop.Excel.XlCellType.xlCellTypeLastCell, Type.Missing);
+            Microsoft.Office.Interop.Excel.Range cellValue = excelSheet.Range["A1", last];
+            
+            List<object> lst = new List<object>();
```
Comment (ob-ab) [inline taskt/Core/Automation/Commands/LoadDictionaryCommand.cs]: **Line - 96**

Extra Line of code (There is no use of this List)
Please remove this.

**Review by ob-ab — COMMENTED** [review]
Please make the requested change.

File: taskt/Core/Automation/Commands/LoadDictionaryCommand.cs [hunk 2]
```diff
@@ -83,15 +84,46 @@ public override void RunCommand(object sender)
...
+            int cCnt;
+
+            DataTable DT = new DataTable();
+
+            for (rCnt = 2; rCnt <= rw; rCnt++)
+            {
+                DataRow newRow = DT.NewRow();
+                for (cCnt = 1; cCnt <= cl; cCnt++)
+                {
+                    if (((cellValue.Cells[rCnt, cCnt] as Microsoft.Office.Interop.Excel.Range).Value2) != null)
```
Comment (melugi) [inline taskt/Core/Automation/Commands/LoadDictionaryCommand.cs]: Can you remove these uses of `Microsoft.Office.Interop.Excel` in the code and just leave the class we want to be referencing?

More generally I think we should be, across the board, removing the direct references to classes like this in a lot of of the code and add them to the using block at the top of each file. We could even go further and move the blocks of using out of the global project namespace and into the class namespace where ever possible.

**Review by melugi — COMMENTED** [review]
Looks good :+1:, I left a comment about about how we're referencing classes and I'd like both your thoughts on it @CameronHerwig @fodfran 

Since we just started on this I think we should make an effort in the areas we touch to keep things a bit neater than we found them and this is a really good start. There's nothing wrong with only taking the parts that we need from a particular namespace, it'll just make the code more succinct. In the future we can make a list of guidelines to check or whatever we need to make this work, but for now this would at least be a good start.

File: taskt/Core/Automation/Commands/LoadDictionaryCommand.cs (hunk 2 above)
Comment (openbots-ff) [reply taskt/Core/Automation/Commands/LoadDictionaryCommand.cs]: Okay, I removed all but the necessary ones.

File: taskt/Core/Automation/Commands/LoadDictionaryCommand.cs (hunk 1 above)
Comment (openbots-ff) [reply taskt/Core/Automation/Commands/LoadDictionaryCommand.cs]: Done

**Review by melugi — APPROVED** [review]
Approved, thanks @fodfran !

File: taskt/Core/Automation/Commands/LoadDictionaryCommand.cs (hunk 2 above)
Comment (CameronHerwig) [reply taskt/Core/Automation/Commands/LoadDictionaryCommand.cs]: I agree. I did a lot of cleanup on the Server component in the same vein. We only need the namespace component when it isnt clear IMO.

**Review by CameronHerwig — APPROVED** [review]
Looks good. Glad we were able to nix the dependency.

**Pull request merged**